From 7a3ac4ac84b406d9d445193a0f21e6dbd85da336 Mon Sep 17 00:00:00 2001
From: Hannes Domani <ssbssa@yahoo.de>
Date: Mon, 2 May 2016 19:31:51 +0200
Subject: [PATCH 8/9] fix printf out of bounds access

If the precision is specified, no more than that many bytes are written.
---
 mingw-w64-crt/stdio/mingw_pformat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mingw-w64-crt/stdio/mingw_pformat.c b/mingw-w64-crt/stdio/mingw_pformat.c
index 021c1eb4d..f0b12c861 100644
--- a/mingw-w64-crt/stdio/mingw_pformat.c
+++ b/mingw-w64-crt/stdio/mingw_pformat.c
@@ -516,7 +516,7 @@ void __pformat_putchars( const char *s, int count, __pformat_t *stream )
       memset(&ps, 0, sizeof(ps) );
       --count;
       p = &w[0];
-      l = mbrtowc (p, s, strlen (s), &ps);
+      l = mbrtowc (p, s, MB_CUR_MAX, &ps);
       if (!l)
         break;
       if ((ssize_t)l < 0)
-- 
2.39.1.windows.1

